iris-mpc-cpu/src/execution/hawk_main/is_match_batch.rs

```diff
@@ -79,20 +79,20 @@ async fn per_session(
 
     let mut out = Vec::with_capacity(tasks.len());
     for (query, vectors) in tasks {
-        let matches = per_query(query, &vectors, &mut session).await;
+        let matches = per_query(&[query], &vectors, &mut session).await;
         out.push(matches);
     }
     out
 }
 
 async fn per_query(
```

Comment:
I guess this name is then kind of misleading now

Reply:
True, but I'd leave this to a follow-up PR.

Reply:
Better move the new `&[query]` inside of `per_query` then.

Reply:
Done